### PR TITLE
fixed bug where adjust_download_permissions was being scheduled on va…

### LINKED ISSUE #32316: adjust_download_permissions row in database
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Reported in 4877582-zen

Whenever the user updates a product (CSV or using the editor), it creates a row `adjust_download_permissions` on the database (even for non-downloadable products).

This brings a problem because they end up with a large database when editing in bulk.

This occurs when updating a variable product without any downloadable variation, but not with a simple product.

### Expected behavior

The `adjust_download_permissions` row in the database should be only added if the product is a downloadable product.

### Actual behavior

Duplicate and multiple `adjust_download_permissions` rows in the database.

### Steps to reproduce

1. Create a new product
2. Select "Variable product"
3. Add attributes
4. Create variations from all attributes

After creating variations I can see the 'adjust_download_permissions' row in the database.

Also after the product is published I can see another 'adjust_download_permissions' row in the database. And the same thing every time I update this product.

So this occurs even if I didn’t create a downloadable product first.

### WordPress Environment

Can reproduce with only WooCommerce + Storefront theme.

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by small-fishes**
…riable products without downloadable variations

### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #32316. First contribution to WooCommerce project.

### How to test the changes in this Pull Request:

1. Create a new product.
2. Select "variable product".
3. Add attributes.
4. Create variations from all attributes. Ensure none are marked as "downloadable". 
5. Edit the product and save again. 
7. Check database or Tools > Scheduled Actions to ensure no "adjust_download_permissions" were scheduled.
8. Mark one or more variations as downloadable. 
9. Edit the product and save again.
10. Check database or Tools > Scheduled Actions to ensure "adjust_download_permissions" was scheduled.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by small-fishes** [comment]
Just wondering, is this waiting on me to do something?

**Comment by seldimi** [comment]
So what happened to this PR, won't it be implemented?

**Comment by small-fishes** [comment]
I'm not sure if I was supposed to do anything further?

File: plugins/woocommerce/src/Internal/DownloadPermissionsAdjuster.php [hunk 1]
```diff
@@ -43,6 +43,19 @@ public function maybe_schedule_adjust_download_permissions( \WC_Product $product
 			return;
 		}
 
+		$are_any_children_downloadable = false;
+		foreach ( $children_ids as $child_id ) {
+			$child = wc_get_product( $child_id );
+			if ( $child->is_downloadable() ) {
```
Comment (therealgilles) [inline plugins/woocommerce/src/Internal/DownloadPermissionsAdjuster.php]: I think you should check that `$child` is not false here. See: https://github.com/woocommerce/woocommerce/pull/35918

Comment (small-fishes) [reply plugins/woocommerce/src/Internal/DownloadPermissionsAdjuster.php]: Thank you. Addressed in commit a97e883c08e059734972076b6708d7e529019011.

**Comment by small-fishes** [comment]
Hi @jorgeatorres can you help me figure out if there is something I've missed that's preventing this PR from being accepted or rejected?

**Comment by jorgeatorres** [comment]
Hi @small-fishes,

We apologize for the delay. We truly appreciate any contributions, but we have tons of open PRs and limited time resources, so it sometimes takes us a while to get to a specific one. I promise we do go through all of them.
Anyways, I will assign this to myself and will be taking a deeper look as soon as possible.

If you're pinging me because I'm reviewing the related PR #35918, that's currently blocked as I'm waiting for a consistent way to reproduce the bug mentioned there. The PR looks good but I just wanted a bit more context in case something more dramatic is going on behind the scenes.

Once again, sorry this has taken so long. And thank you for contributing.

**Comment by seldimi** [comment]
If I could help, I can see the issue when creating products with CRUD. iF you create/update a product (loaded with new WC_product_variable() or wc_get_product(XXXX), do some CRUD like $product->set_stock_quantity(10); and then $product->save(), you will see the action scheduler task to popup on the queue, even if the product is not digital (eg a t-shhirt)

**Review by jorgeatorres — DISMISSED** [review]
LGTM.

**Comment by jorgeatorres** [comment]
Hi @small-fishes!

Thanks for your contribution 💯. Sorry this took so long to get reviewed, but it's finally approved. I'm waiting on some checks to finish and will merge afterwards.
Thanks again!

**Review by jorgeatorres — APPROVED** [review]

**Pull request merged**